### ISSUE TITLE
Local plugins not detected when using workspace.json 

### ISSUE THREAD

**Issue by robdonn** [nrwl/nx#12451]
## Current Behavior

When using the latest version of NX (14.8.3 as of this issue creation) commands fail as NX cannot resolve the location of a local nx-plugin if a `workspace.json` file is used in the project.

## Expected Behavior

NX should be able to resolve the location of the local nx-plugin.

## Steps to Reproduce

[Example PR](https://github.com/nrwl/nx-examples/pull/236)

- Create an NX project
- Generate an app (e.g. React app)
- Generate a local nx-plugin (`@nrwl/nx-plugin`)
- In `nx.json` add the local nx-plugin to the `plugins` field.
- Create a `workspace.json` at the root of the project
  - `{ "version": 2, "projects: [] }`
- `nx run <appName>:build`

### Failure Logs

Log from [example pull request](https://github.com/nrwl/nx-examples/pull/236) in `nx-examples` repo.

```sh
 >  NX   Cannot find module '@nx-example/broken-plugin'

   Require stack:
   - /Users/robert.donnelly/robdonn/nx-examples/node_modules/nx/src/utils/nx-plugin.js
   - /Users/robert.donnelly/robdonn/nx-examples/node_modules/nx/src/config/workspaces.js
   - /Users/robert.donnelly/robdonn/nx-examples/node_modules/nx/src/config/configuration.js
   - /Users/robert.donnelly/robdonn/nx-examples/node_modules/nx/src/daemon/server/project-graph-incremental-recomputation.js
   - /Users/robert.donnelly/robdonn/nx-examples/node_modules/nx/src/daemon/server/server.js
   - /Users/robert.donnelly/robdonn/nx-examples/node_modules/nx/src/daemon/server/start.js
   
   Because of the error the Nx daemon process has exited. The next Nx command is going to restart the daemon process.
   If the error persists, please run "nx reset".
```

### Environment

> Note: I had to remove the local plugin reference from `nx.json` to get `nx report` command to work

```sh
   Node : 16.10.0
   OS   : darwin arm64
   yarn : 1.22.11
   
   nx : 14.8.3
   @nrwl/angular : 14.8.3
   @nrwl/cypress : 14.8.3
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.8.3
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 14.8.3
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 14.8.3
   @nrwl/js : 14.8.3
   @nrwl/linter : 14.8.3
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : 14.6.1
   @nrwl/nx-plugin : 14.8.3
   @nrwl/react : 14.8.3
   @nrwl/react-native : Not Found
   @nrwl/rollup : 14.8.3
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.8.3
   @nrwl/web : 14.8.3
   @nrwl/webpack : 14.8.3
   @nrwl/workspace : 14.8.3
   typescript : 4.7.2
   ---------------------------------------
   Local workspace plugins:
         @nx-example/broken-plugin
   ---------------------------------------
   Community plugins:
         @ngrx/component-store: 14.0.2
         @ngrx/effects: 14.0.2
         @ngrx/entity: 14.0.2
         @ngrx/router-store: 14.0.2
         @ngrx/store: 14.0.2
         @ngrx/store-devtools: 14.0.2
```

**Comment by lewsmith** [comment]
I'm also seeing an issue with 14.8.3 and local plugins. Slightly different to above.

I'm using a plugin to extend the dep graph for go projects. It was working perfectly with the previous version but now I'm getting the following error when trying to run `nx graph`:

```sh
> NX   unable to find tsconfig.base.json or tsconfig.json
```
https://github.com/nrwl/nx/blob/9b63ce167acb08faaf55bdf899055ae04a87036f/packages/nx/src/utils/nx-plugin.ts#L233-L245

The latest version appears to be searching for tsconfig files in every project. However, I have go projects that do not have a tsconfig in it's project root. I don't really want to add one either. For now I will roll back to the previous version, but thought I'd highlight the issue here.

**Comment by robdonn** [comment]
Closing as my issue appears to be a duplicate of #12340 which was fixed by #12406 

I replicated the change locally and it did fix the problem. Just waiting for it to make it into a release now.
